src/text-input/src/TextInput.js

```diff
@@ -89,25 +118,55 @@ class TextInput extends PureComponent {
     const borderRadius = theme.getBorderRadiusForControlHeight(height)
 
     return (
-      <Text
-        is="input"
-        className={cx(themedClassName, className)}
-        type="text"
-        size={textSize}
-        width={width}
-        height={height}
-        required={required}
-        disabled={disabled}
-        placeholder={placeholder}
-        paddingLeft={Math.round(height / 3.2)}
-        paddingRight={Math.round(height / 3.2)}
-        borderRadius={borderRadius}
-        spellCheck={spellCheck}
-        aria-invalid={isInvalid}
-        {...(disabled ? { color: 'muted' } : {})}
-        css={css}
-        {...props}
-      />
+      <React.Fragment>
+        <Text
+          is="input"
+          className={cx(themedClassName, className)}
+          type="text"
+          size={textSize}
+          width={width}
+          height={height}
+          required={required}
+          disabled={disabled}
+          placeholder={placeholder}
+          value={this.getValue()}
+          {...(onChange
+            ? { onChange }
+            : { onChange: e => this.setState({ value: e.target.value }) })}
```

Comment:
I'd recommend that we always call our own function, and only if the consumer also passes an `onChange` prop do we bubble to them as well (you can use `safeInvoke` for this):

```js
class TextInput extends PureComponent {
  handleChange = (event) => {
    this.setState({ value: e.target.value })
    safeInvoke(this.props.onChange, event)
  }

  render() {
    // later...
    <Text
      onChange={this.handleChange}
    />
  }
}
```